aws-api-appsync/src/main/java/com/amplifyframework/api/aws/GraphQLRequestHelper.java

```diff
@@ -224,21 +226,30 @@ private static Map<String, Object> extractFieldLevelData(
                 continue;
             }
 
-            Object fieldValue = extractFieldValue(modelField.getName(), instance, schema);
+            Object fieldValue = extractFieldValue(modelField.getName(), instance, schema, false);
+            Object underlyingFieldValue = fieldValue;
+            if (modelField.isModelReference() && fieldValue != null) {
+                ModelReference<?> modelReference = (ModelReference<?>) fieldValue;
+                if (modelReference instanceof LoadedModelReference) {
+                    underlyingFieldValue = ((LoadedModelReference) modelReference).getValue();
+                }
+            }
 
             if (association == null) {
                 result.put(fieldName, fieldValue);
             } else if (association.isOwner()) {
-                if (fieldValue == null && MutationType.CREATE.equals(type)) {
+                if ((fieldValue == null ||
+                        (modelField.isModelReference() && underlyingFieldValue == null)) &&
+                        MutationType.CREATE.equals(type)) {
                     // Do not set null values on associations for create mutations.
```

Comment:
Correct, but its more of a stylistic preference as behavior is identical. I've mixed up `continue` and `break` before so tend to prefer omitting continue entirely to prevent that accident.